lib/api/controllers/adminController.js

```diff
@@ -24,7 +24,7 @@ module.exports = function AdminController (kuzzle) {
         kuzzle.indexCache.add(modifiedRequestObject.index, modifiedRequestObject.collection);
         return kuzzle.pluginsManager.trigger('data:afterUpdateMapping', new ResponseObject(modifiedRequestObject, response));
       })
-      .catch(err => q.reject(new ResponseObject(modifiedRequestObject || requestObject, err)));
+      .catch(err => q.reject(err));
```

Comment:
This line, and all lines alike, should be removed. The rejected promise is already forwarded to the caller, we intercepted errors to reject a ResponseObject instead of the original error.